Mesh4all/m4ag-firmware
# ci: add static tests and compilation validation

### Contribution description

This PR adds some test and tools for CI and code quality

### Testing procedure

Some test will run as GH actions, is expected that locally can be executing the following command in the root folder: 
```sh
make static-test
```

The compilation check try to compile for the hardware reference, using the following command:

```sh
make -C firmware BOARD=m4a-mb
```

### Issues/PRs references

This PR fix #6
